app/src/main/java/com/ncnf/map/ui/MapFragment.java

```diff
@@ -58,13 +66,21 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
 
         View view = inflater.inflate(R.layout.fragment_map, container, false);
 
+        requestPermissionLauncher = registerForActivityResult(new ActivityResultContracts.RequestPermission(), isGranted -> {
+            if (isGranted) {
+                returnToLocation(view);
+            }
+        });
+
```

Comment:
What happens if the user refuses the permission ?

Reply:
Nothing, the function is not called and the position remains the same